### ISSUE TITLE
In Ubuntu "Default" radio buttons not selected, instead custom text fields have default values

### ISSUE THREAD

**Issue by lunarovic** [webmin/webmin#910]
In Ubuntu (clean install), a lot of the settings in the modules have values written in the custom box, even though that's the default value and the radio option should be selected instead. In CentOS, all those settings just have the "Default" or "Automatic" radio option selected, which is normal.

In CentOS even when the radio option is "None", in Ubuntu the custom box is selected instead and it's empty. And there are even things like instead of the radio option "All" being selected, the word "all" is written in the custom field.

And if I try changing for example the option from an empty box to "None", it will not save and it will return back to the empty box. So those "None" (or "Not set" etc.) radio buttons can practically never be selected and used, so they're kinda useless.

I'm concerned that this is bad for user experience (confusion), because I don't know what is the default value. Is that value in the box different than the default or not? So system admins can't easily find out what is changed from their default values.

Some examples to better illustrate what I'm talking about:

![image](https://user-images.githubusercontent.com/2381553/39396330-8672dbbe-4aec-11e8-9003-bc25f53ed35c.png)

or

![image](https://user-images.githubusercontent.com/2381553/39396341-cbfd4ea8-4aec-11e8-8bfb-d8cddc4ccecb.png)

or

![image](https://user-images.githubusercontent.com/2381553/39396346-e0d7f832-4aec-11e8-91f4-4328470351f3.png)

or

![image](https://user-images.githubusercontent.com/2381553/39396417-048bd234-4aee-11e8-8fde-709cedfc68bb.png)

or

![image](https://user-images.githubusercontent.com/2381553/39396370-4a2c3122-4aed-11e8-9378-883fcf506032.png)

I can show many examples like this, this is across all Webmin.

These screenshots are from Ubuntu. In CentOS, the first radio button would always be selected in such cases.

**Comment by qooob** [comment]
Please be kind reporting it to [Webmin repo.](https://github.com/webmin/webmin/)

**Comment by jcameron** [comment]
That's unusual .. which browser are you using there?

**Comment by lunarovic** [comment]
Chrome 66 (64-bit) on Windows 10. And it's the same in every Webmin theme.

**Comment by jcameron** [comment]
Actually, this may be a server-side issue. Which Ubuntu version are you running?

**Comment by lunarovic** [comment]
I think so too, because that does not happen on CentOS in the same environment.

I just noticed that Webmin System Information shows `Ubuntu 16.04.3`, while when I type `lsb_release -a` in the terminal, I get `Ubuntu 16.04.4`.

Kernel: `Linux 4.4.0-93-generic on x86_64`

Processor information: `QEMU Virtual CPU version (cpu64-rhel6), 1 cores`

Webmin version: `1.881`

**Comment by jcameron** [comment]
Actually, I think this is expected - in the Webmin UI, the "default" box is only selected when the config file is missing a setting. If it's in the config but still set to whatever the default would be if it was missing, what you saw is what the UI will display.

**Comment by lunarovic** [comment]
You're talking about Ubuntu, right? Because on CentOS the first radio button would always be selected in those cases where the value is initial, i.e. it hasn't changed from the default.

So you're saying that Ubuntu explicitly sets the default values in the config files, so that's what's being read by Webmin, while CentOS does not set anything explicitly, so it falls back to defaults?

If so, how do you explain the "None" or "No map set" cases (see screenshots above), where the empty text box is selected instead - surely that can't be a config setting?

Also, if I switch the radio button from the empty text box to "None", it will not save and it will return back to the empty text box. Isn't that evidence that those two are not separate things?

**Comment by jcameron** [comment]
Yes, Ubuntu appears to explicitly set the defaults in the config file.

However, a text box being empty and selected should never happen. What does the `always_bcc` line in your `/etc/postfix/main.cf` file contain?

**Comment by lunarovic** [comment]
I don't even have that line in main.cf. Only this:
`sender_bcc_maps = hash:/etc/postfix/bcc`

And that bcc file is empty.

**Comment by jcameron** [comment]
What output do you get if you run the command `postconf -h always_bcc`

**Comment by lunarovic** [comment]
```
# postconf -h always_bcc
postconf: warning: /etc/postfix/master.cf: undefined parameter: mua_sender_restrictions
postconf: warning: /etc/postfix/master.cf: undefined parameter: mua_client_restrictions
postconf: warning: /etc/postfix/master.cf: undefined parameter: mua_helo_restrictions
```

But when I run just `postconf` and scroll in the list to find `always_bcc`, I see this:

```
alternate_config_directories =
always_add_missing_headers = no
always_bcc =
anvil_rate_time_unit = 60s
anvil_status_update_time = 600s
```

There is no space or any other character after the equals sign, just a new line.

And if I try it with some parameter which has data, then I would get this:
```
# postconf -h anvil_rate_time_unit
postconf: warning: /etc/postfix/master.cf: undefined parameter: mua_sender_restrictions
postconf: warning: /etc/postfix/master.cf: undefined parameter: mua_client_restrictions
postconf: warning: /etc/postfix/master.cf: undefined parameter: mua_helo_restrictions
60s
```

**Comment by lunarovic** [comment]
There are no those parameters in main.cf, but I see they are in master.cf:
```
smtps	inet	n	-	y	-	-	smtpd -o syslog_name=postfix/smtps -o
smtpd_tls_wrappermode=yes -o smtpd_sasl_auth_enable=yes -o
smtpd_reject_unlisted_recipient=no -o
smtpd_client_restrictions=$mua_client_restrictions -o
smtpd_helo_restrictions=$mua_helo_restrictions -o
smtpd_sender_restrictions=$mua_sender_restrictions -o
smtpd_recipient_restrictions= -o
smtpd_relay_restrictions=permit_sasl_authenticated,reject -o
milter_macro_daemon_name=ORIGINATING
```

If I remove those 3 lines and reload Postfix, there are no those errors anymore, but it's still the same with the selected empty text fields.

But I'm pretty sure this doesn't have anything to do with Postfix, this condition is across all Webmin on Ubuntu, in all settings everywhere. I can make more screenshots from any module you want, just tell me which one.

**Comment by jcameron** [comment]
Yes, please post a screenshot  from any other module where you're seeing this (text box selected, but empty).

**Comment by lunarovic** [comment]
Turns out that it seems the selected empty box does not manifest in other modules except Postfix - in my mind I put it together with the "default" issue, which does happen across different modules. Nevertheless, here are a few more interesting screenshots from Postfix settings:

Here the first box is filled and the other one is empty, and "No map set" is not selected:
![image](https://user-images.githubusercontent.com/2381553/39751507-b2cd0288-52b8-11e8-9f82-6401e278fbc9.png)

Here is the whole `postfix/local_delivery.cgi` page, there are multiple occurrences there:
![image](https://user-images.githubusercontent.com/2381553/39751584-e5f11410-52b8-11e8-8665-0d46178452b9.png)
I'm wondering does this have also to do with the defaults being different in CentOS and Ubuntu? Because the text on the user mailbox file setting says that `/var/something` is default, when in Ubuntu it seems to be `Maildir/`?

And here is a strange case outside of Postfix where nothing is selected:
![image](https://user-images.githubusercontent.com/2381553/39751148-902cccdc-52b7-11e8-920d-b1f34e87ad2a.png)

So aside from these "empty" issues, in around third of all module settings I see default values being written in the text field instead of the radio "Default" being selected, which I still think is an issue for Webmin on Ubuntu. The other two thirds of settings have the "Default" option selected correctly though, depending on the module.

So maybe it could be found what exactly causes a Webmin module to read a parameter and say "this is a default value, I will check the default radio button" compared to "this is different than default, this is an explicit custom value". Or Ubuntu sets different parameters in different ways (leave unset / set explicitly)?

See this example in `edit_plan.cgi`:
![image](https://user-images.githubusercontent.com/2381553/39750721-43359b62-52b6-11e8-8305-66d2d7081688.png)
Everything there are default/initial values, I didn't change anything. So why is the limit on number of virtual servers being read differently than all of the other settings, when it's obviously the same?

Also, a related issue is that I see inconsistency in the disabled text fields across Webmin, and Ilia says that's on the server side. For clarification please take a look here: https://github.com/qooob/authentic-theme/issues/1130

Thank you for your time.

**Comment by iliaross** [comment]
❗️ If the issue isn't resolved, feel free to reopen it with updated details.
